prov/sockets/src/sock_cq.c

```diff
@@ -368,6 +368,8 @@ static ssize_t sock_cq_sreadfrom(struct fid_cq *cq, void *buf, size_t count,
 				ret = sock_cq_rbuf_read(sock_cq, buf,
 					MIN(threshold, avail / cq_entry_len),
 					src_addr, cq_entry_len);
+			else /* No CQ entry available, read the fd */
+				rbfdreset(&sock_cq->cq_rbfd);
```

Comment:
The UDP provider took a different approach that may be useful here.  Whenever an error entry is added to the error queue, a matching entry is added to the normal CQ ring buffer with an error flag set.  When attempting to read from the CQ, only the main ring buffer is checked, and if the error flag is set, the CQ read returns EAVAIL. The error data can then be read from the error queue.  This keeps error entries and non-error entries ordered, and only the sread call needs to deal with the error queue.  All other code uses the ring buffer as normal.